content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md

```diff
@@ -0,0 +1,486 @@
+          "__meta_kubernetes_service_label_app": "my-app",
+          "__meta_kubernetes_endpointslice_labelpresent_app_kubernetes_io_name": "true"
+        }
+      }
+    ]
+  }
+}
+```
+
+{{% alert title="Note" %}}
+
+See the
+[Target Allocator readme](https://github.com/open-telemetry/opentelemetry-operator/blob/main/cmd/otel-allocator/README.md?plain=1#L128-L134)
+for more information on the `/jobs` endpoint.
+
+{{% /alert %}}
+
+### Is the Target Allocator enabled? Is Prometheus service discovery enabled?
+
+If the `curl` commands above don’t show a list of expected `ServiceMonitor`s and
+`PodMonitor`s, you need to check whether the features that populate those values
+are turned on.
+
+One thing to remember is that just because you include the `targetAllocator`
+section in the `OpenTelemetryCollector` CR doesn’t mean that it’s enabled. You
+need to explicitly enable it. Furthermore, if you want to use
+[Prometheus service discovery](https://github.com/open-telemetry/opentelemetry-operator/blob/main/cmd/otel-allocator/README.md#discovery-of-prometheus-custom-resources),
+you must explicitly enable it:
+
+- Set `spec.targetAllocator.enabled` to `true`
+- Set `spec.targetAllocator.prometheusCR.enabled` to `true`
+
+So that your `OpenTelemetryCollector` resource looks like this:
+
+```yaml
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: otelcol
+  namespace: opentelemetry
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+    serviceAccount: opentelemetry-targetallocator-sa
+    prometheusCR:
+      enabled: true
+```
+
+See the full `OpenTelemetryCollector`
+[resource definition in "Do you know if metrics are actually being scraped?"](#do-you-know-if-metrics-are-actually-beingscraped).
+
+### Did you configure a ServiceMonitor (or PodMonitor) selector?
+
+If you configured a
+[`ServiceMonitor`](https://observability.thomasriley.co.uk/prometheus/configuring-prometheus/using-service-monitors/#:~:text=The%20ServiceMonitor%20is%20used%20to,build%20the%20required%20Prometheus%20configuration.)
+selector, it means that the Target Allocator only looks for `ServiceMonitors`
+having a `metadata.label` that matches the value in
+[`serviceMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr).
+
+Suppose that you configured a
+[`serviceMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr)
+for your Target Allocator, like in the following example:
+
+```yaml
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: otelcol
+  namespace: opentelemetry
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+    serviceAccount: opentelemetry-targetallocator-sa
+    prometheusCR:
+      enabled: true
+      serviceMonitorSelector:
+        app: my-app
```

Comment:
This would be correct in apiVersion `v1alpha1`, but in `v1beta1` this field is a label selector, and should look like this:

```suggestion
      serviceMonitorSelector:
        matchLabels:
          app: my-app
```

You also have the documentation for v1alpha1 linked above, which is most likely the source of the confusion here. The v1beta1 struct is documented [here](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr-1).

Reply:
Oooh, thanks for that catch!